### PR TITLE
fix: production-merge gaps — GRPO on HF Jobs, Trackio, reward defaults

### PR DESCRIPTION
**Pull request by GunaPalanivel**
## Summary
Housekeeping PR for production-merge checklist: reward signal, sane GRPO defaults, Trackio-only logging, HF Jobs launcher with clone-race preflight, and trainer dependency/runtime fixes so a full GPU GRPO job reaches the training loop.

## Root causes addressed
- **P0/P1:** fficiency_bonus_max too low / missing positive signal; default LR and group size misaligned with TRL GRPO expectations.
- **Deps:** TRL (as pinned by Unsloth) pulled optional imports (mergekit, llm-blender); llm-blender + Transformers 5.x broke on removed TRANSFORMERS_CACHE — pinned 	ransformers<5 within Unsloth’s allowed range.
- **Runtime:** generation_batch_size default incompatible with 
um_generations=8; 4-bit model needed Unsloth LoRA before GRPOTrainer fine-tuning.

## Verification
- \pytest tests/ -q\: **498 passed** (local).
- **HF Jobs**
  - [\69edcfdad2c8bd8662bcfa07\](https://huggingface.co/jobs/gp5901/69edcfdad2c8bd8662bcfa07): **completed — failed** at import (\mergekit\ missing); documents the failure mode we fixed.
  - [\69edd94dd2c8bd8662bcfb08\](https://huggingface.co/jobs/gp5901/69edd94dd2c8bd8662bcfb08): **validates** full stack (Trackio + Hub + GRPO steps through 200-step schedule).

## Judge links (stable)
- **Trackio (live):** https://gp5901-trackio.hf.space/
- **Trackio Space:** https://huggingface.co/spaces/gp5901/trackio
- **Hub model:** https://huggingface.co/gp5901/praxis-grpo-7b

## How to re-run
\\\ash
python scripts/submit_hf_grpo_job.py --hub-model-id gp5901/praxis-grpo-7b --trackio-space-id gp5901/trackio --flavor a10g-large --timeout 5h
\\\

Link your tracking issue in a follow-up comment if the number differs from the merge checklist doc.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by Gokul287 — APPROVED** [review]
Local verification on 524a38f: ruff format/check, openenv validate, pytest (3.12 dedicated venv + default env), HTTP /reset smoke. CI green on head.
